### ISSUE TITLE
Improve Incident and Scheduled Maintenance Systems

### ISSUE THREAD

**Issue by jbrooksuk** [CachetHQ/Cachet#1176]
I'm creating this issue to group previous issues into one place, because they're all related:
- [x] https://github.com/cachethq/Cachet/issues/530 - Incident timeline of updates
- [x] https://github.com/cachethq/Cachet/issues/754 - Choose when to display Scheduled Maintenances.
- [x] https://github.com/cachethq/Cachet/issues/715 - Improving Scheduled Maintenance
- [ ] https://github.com/cachethq/Cachet/issues/495 - Scheduled Maintenance should affect (multiple) components
- [ ] https://github.com/cachethq/Cachet/issues/387 - Incidents should affect multiple components
- [ ] https://github.com/CachetHQ/Cachet/issues/668 - Incidents should be reportable on a component group
- [ ] https://github.com/CachetHQ/Cachet/issues/674 - Recurring maintenance
- [ ] Incident and Schedule drafting, i.e. write but don't publish.

**Comment by jbrooksuk** [comment]
I've scrubbed off #674 because with incident updates this will make it easier to work out the duration of an incident :+1:

**Comment by jbennett999** [comment]
Just want to throw in a +1 for #530 - Incident timeline of updates

It's the one part of Cachet that throws me off. I find myself switching between updating old incidents with status updates/component changes and creating "new" incidents to update the old ones which artificially inflates my incident count and makes it hard to follow individual incidents.

I'd like the timeline to reflect the work being done so you can see an incident go from Investigating -> Identified -> Watching -> Fixed and the related component statuses. When you update the original incident, it removes the historical data.

[Google's Method](https://www.google.com/appsstatus#hl=en&v=issue&sid=4&iid=fb887d4953c88406aab04e737d583560)
[Staytus' Method](https://camo.githubusercontent.com/469d676d5562c5d379786a4af2793cde9e5f6ed4/68747470733a2f2f732e6164616d636f6f6b652e696f2f31352f4a4572584537354668752e706e67)

Just my 2¢. Thanks for a great project!

**Comment by jbrooksuk** [comment]
👍

**Comment by jbrooksuk** [comment]
We're working on a new design to simplify the incident updates too.

**Comment by rayterrill** [comment]
Trying to follow along with the issues - we saw the issue for maintenance durations - does this mean it's now coming in v3 or v4 of Cachet? Any ETA for that release? Just getting started with Cachet and it looks pretty rad so far. :smile:

**Comment by jbrooksuk** [comment]
@rayterrill thanks! Can you point me to the issue for maintenance durations?

**Comment by rayterrill** [comment]
@jbrooksuk I think it was this issue here - where you modified the milestone here: "jbrooksuk modified the milestone: V3.0.0, V4.0.0 on Apr 21, 2016".

**Comment by jbrooksuk** [comment]
It's ongoing 😉 progress has been slow for a while as my full-time job has been taking a lot of my free time too.

**Comment by c0ri** [comment]
@jbrooksuk Hi James, is it possible to also add a list of approvers for said maintenance?   I imagine Cachet could send email to people's email addresses you list in a maint. req. and then it sends email with a link. When those people click the link it would show they've approved of the change on the Scheduled maint. page.

Since you are redoing this, thought I'd throw that in there :)  Thanks!

**Comment by jbrooksuk** [comment]
I think instead of that, we should be able to draft maintenance...

**Comment by jbrooksuk** [comment]
And incidents for that matter.

**Comment by c0ri** [comment]
@jbrooksuk Can you elaborate on your meaning of drafting maintenance?  I see a lot of new actions in the API. I've added those to my code but they don't work yet :P

**Comment by jbrooksuk** [comment]
So basically, you can start writing a new incident or maintenance but not publish it, so you can come back to it later. Especially useful for maintenance where you may want to start writing it but don't have all of the facts yet.

**Comment by MrFloppy** [comment]
Which issues are open for v3? Maybe I can help you out?

**Comment by jbrooksuk** [comment]
Any non-ticked items above @MrFloppy :)

**Comment by pendraggon87** [comment]
Any reason #674 was scrubbed?  Being able to schedule a recurring maintenance window would be huge, and is one of the biggest features that many individuals would need.  We have people that constantly report an app is down even though it is during a maintenance window.

Love the dashboard :)

**Comment by jbrooksuk** [comment]
@pendraggon87 to be honest, I'm not sure... In #674 I said it was merged into this issue, so it should still be there. Will add it back.
